pkg/diagnostics/pod/dns.go

```diff
@@ -101,6 +105,7 @@ func resolveSearch(resolvConf *dns.ClientConfig, r types.DiagnosticResult) {
 		return string(b)
 	}()
 	for _, domain := range resolvConf.Search {
+		// TODO: use dnsDomain instead of cluster.local
```

Comment:
I believe the correct way to check this will be to have the container make a DNS query to `kubernetes.default` and observe the resolved name.